rust/datafusion/src/logicalplan.rs

```diff
@@ -190,7 +190,14 @@ fn create_name(e: &Expr, input_schema: &Schema) -> Result<String> {
             let expr = create_name(expr, input_schema)?;
             Ok(format!("CAST({} as {:?})", expr, data_type))
         }
-        Expr::ScalarFunction { name, args, .. } => {
+        Expr::ScalarFunction { fun, args, .. } => {
+            let mut names = Vec::with_capacity(args.len());
```

Comment:
I don't know how much you care about using "idomatic" rust, but you can probably do the same kind of thing here with iterators:

```
let names = args.iter()
  .map(|e| create_name(e, input_schema)
  .collect::<Result<Vec_>>()?;
```

Reply:
I agree. I kept it because the code around this change is using this idiom. I would prefer to change that whole code (`ScalarUDF`, `AggregateFunction`, `ScalarFunction`) in a separate commit to not want to complicate even further this PR

Reply:
that makes sense